functions/go/set-image/image_transformer_test.go

```diff
@@ -0,0 +1,271 @@
+package main
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"sigs.k8s.io/kustomize/kyaml/fn/sdk"
+)
+
+// helper function to convert ResourceList items to yaml
+func itemsToYaml(items []*sdk.KubeObject) string {
+	var itemYamls []string
+	for _, item := range items {
+		itemYamls = append(itemYamls, item.String())
+	}
+	result := strings.Join(itemYamls, "---\n")
+	return result
+}
+
+func runImageTransformerE(input, config string) (string, error) {
```

Comment:
@mengqiy Let me know if there is a better way to handle this unit test scaffolding in the new SDK.

Given this is along the right track, I was thinking we might consider implementing a similar layer in the SDK to streamline the experience of writing tests and reduce code duplication.

Reply:
Have you looked at http://104.197.156.154/pkg/sigs.k8s.io/kustomize/kyaml/fn/sdk/testutil/?
That package currently have 3 methods and they assume that golden tests will be used. Maybe we should add another helper that take inline input.

Reply:
Yeah I did take a look at the `testutil` package

I felt that creating a separate file/directory for each test case would become a bit cumbersome in this instance, so I opted to create this function which accepts string input instead. I was thinking as you say that we could add something like this as another helper which accepts inline (string) input. WDYT?